ui/src/app/views/workspace-edit.tsx

```diff
@@ -390,8 +434,45 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
         cloneUserRole: false,
         loading: false,
         showUnderservedPopulationDetails: false,
-        showStigmatizationDetails: false
+        showStigmatizationDetails: false,
+        autocompleteLoading: false,
+        autocompleteDiseases: [],
+        diseaseHover: [],
+        dropDown: false,
+        searchTerm: '',
+        hovering: false
       };
+      this.searchTermChangedEvent = fp.debounce(300, this.diseaseSearch);
+    }
+
+    get showSearchResults(): boolean {
+      return !this.state.autocompleteLoading &&
+        this.state.autocompleteDiseases.length > 0 &&
+        !isBlank(this.state.searchTerm) && this.state.dropDown;
+    }
+
+    diseaseSearch(value: string): void {
+      this.setState({autocompleteLoading: true, autocompleteDiseases: [], searchTerm: value});
+      if (!value.trim()) {
+        this.setState({autocompleteLoading: false, dropDown: false});
```

Comment:
how about writing this as         this.setState({autocompleteLoading: value.trim.....});

Reply:
good idea. i was able to remove a bunch of duplicate state.